### PR TITLE
Move lastAppProps back where it should be

### LINKED ISSUE #1061: TypeError: Cannot read property 'Component' of undefined
I am using the [cloudinary-react library](https://github.com/cloudinary/cloudinary-react) and have run into the following error message when I add the Image component to my project:

**In the DOM**
```plaintext
TypeError: Cannot read property 'Component' of undefined
    at _callee3$ (http://localhost:3000/_next/-/main.js:3977:50)
    at tryCatch (http://localhost:3000/_next/-/commons.js:36939:40)
    at Generator.invoke [as _invoke] (http://localhost:3000/_next/-/commons.js:37230:22)
    at Generator.prototype.(anonymous function) [as next] (http://localhost:3000/_next/-/commons.js:36991:21)
    at step (http://localhost:3000/_next/-/commons.js:3724:30)
    at http://localhost:3000/_next/-/commons.js:3742:14
    at F (http://localhost:3000/_next/-/commons.js:2033:28)
    at http://localhost:3000/_next/-/commons.js:3721:12
    at doRender (http://localhost:3000/_next/-/main.js:3999:18)
    at Object._callee$ (http://localhost:3000/_next/-/main.js:3891:20)
```

**In the devtools**
```plaintext
Warning: React attempted to reuse markup in a container but the checksum was invalid. 
This generally means that you are using server rendering and the markup generated on the server was not what the client was expecting. 
React injected new markup to compensate which works but you have lost many of the benefits of server rendering. 
Instead, figure out why the markup being generated is different on the client or server:
 (client) tid="1"><div class="error" data-jsx="676
 (server) tid="1"><div class="wrapper" data-jsx="2
```

The Image component takes in a `cloudName` & `publicId`. The `publicId` is a string that represents a photo on the Cloudinary API server attached to the user's `cloudName`. Could it be that this is CORS issue and causing this error to appear? I'm not entirely sure what this is saying.

### PR DESCRIPTION
**Pull request by timneutkens**
Fixes #1061

### CODE REVIEW

File: client/index.js [hunk 1]
```diff
@@ -77,11 +77,10 @@ async function doRender ({ Component, props, err, emitter }) {
   props = props || lastAppProps.props
 
   const appProps = { Component, props, err, router, headManager }
+  lastAppProps = appProps
```
Comment (nkzawa) [inline client/index.js]: Can you add a comment which this has to be before `render` ?

Comment (timneutkens) [inline client/index.js]: Sure!

Comment (timneutkens) [inline client/index.js]: Done 👍

**Comment by nkzawa** [comment]
Thanks!

**Comment by rauchg** [comment]
Good stuff @timneutkens

**Pull request merged**